rtengine/simpleprocess.cc

```diff
@@ -1369,41 +1369,75 @@ class ImageProcessor {
             params.crop.enabled = false;
         }
 
-        if (params.resize.enabled) {
-            // resize image
-            LabImage *resized = new LabImage(imw, imh);
-            ipf.Lanczos(tmplab, resized, tmpScale);
-            delete tmplab;
-            tmplab = resized;
-            params.resize.enabled = false;
+        assert(params.resize.enabled);
 
-            params.sharpening = params.prsharpening;
-            params.impulseDenoise.thresh = 
-                int(float(params.impulseDenoise.thresh) * tmpScale);
-            if (tmpScale < 0.5) {
-                params.impulseDenoise.enabled = false;
-            }
-            params.wavelet.strength =
-                int(float(params.wavelet.strength) * tmpScale);
-            params.dirpyrDenoise.luma *= tmpScale;
-            //params.dirpyrDenoise.smethod = "shal";
-
-            const double dirpyreq_scale = min(tmpScale * 1.5, 1.0);
-            for (int i = 0; i < 6; ++i) {
-                auto &n = params.dirpyrequalizer.mult[i];
-                n = 1.0 + (n - 1.0) * dirpyreq_scale;
-            }
+        // resize image
+        LabImage *resized = new LabImage(imw, imh);
+        ipf.Lanczos(tmplab, resized, tmpScale);
+        delete tmplab;
+        tmplab = resized;
+
+        adjust_procparams(tmpScale);
             
-            fw = imw;
-            fh = imh;
-        }
+        fw = imw;
+        fh = imh;
 
         delete baseImg;
         baseImg = new Imagefloat(fw, fh);
         ipf.lab2rgb(*tmplab, *baseImg, params.icm.working);
         delete tmplab;
     }
 
+    void adjust_procparams(double scale_factor)
+    {
+        procparams::ProcParams& params = job->pparams;
+        procparams::ProcParams defaultparams;
+
+#define ADJUST_RADIUS_(n, f)                                              \
```

Comment:
I would be happy to, but I'm afraid I will need some help - my knowledge of C++1x is probably too low to come up with a concise and readable solution. Do you have any suggestions?